File: api/Global.go
package api

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Global 对应 Global.py。
type Global struct {
	*session.ApiGroup
}

// NewGlobal 创建 Global API 分组。
func NewGlobal(s *session.Session) *Global {
	return &Global{ApiGroup: session.NewApiGroup(s)}
}

// ModuleSwitch 对应 global/module-switch。
func (g *Global) ModuleSwitch() (map[string]interface{}, error) {
	return g.S.Get("global/module-switch", nil, "api")
}

// StorageGetItem 对应 global/storage-getitem。
func (g *Global) StorageGetItem() (map[string]interface{}, error) {
	return g.S.Get("global/storage-getitem", nil, "api")
}

// CustommenuUrl 对应 global/custommenu-url。
func (g *Global) CustommenuUrl() (map[string]interface{}, error) {
	return g.S.Get("global/custommenu-url", nil, "api")
}

File: src/api/global.rs
//! Global API group (`api/Global.py`).

use serde_json::Value;

use crate::connection::Connection;
use crate::errors::Result;

use super::get_value;

/// Global API group.
pub struct Global<'a> {
    conn: &'a Connection,
}

impl<'a> Global<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        Global { conn }
    }

    /// `global/module-switch`. Module switch status.
    pub fn module_switch(&self) -> Result<Value> {
        get_value(self.conn, "api/global/module-switch")
    }

    /// `global/storage-getitem`. Storage item (reverse engineered).
    pub fn storage_get_item(&self) -> Result<Value> {
        get_value(self.conn, "api/global/storage-getitem")
    }

    /// `global/custommenu-url`. Custom menu URL.
    pub fn custommenu_url(&self) -> Result<Value> {
        get_value(self.conn, "api/global/custommenu-url")
    }
}
